### PR TITLE
Make txselector ignore exits with amount 0

### PR DESCRIPTION
**Pull request by arnaubennassar**
**EDIT:** Repurposed the PR to filter this txs a txselector level

### What does this PR does?

Make txselector ignore transactions of type exit with amount 0

## Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Include tests
- [ ] Respect code style and lint
- [ ] Update swagger file (if needed)
- [ ] Update documentation (*.md) (if needed)

### CODE REVIEW

File: txprocessor/txprocessor.go [hunk 1]
```diff
@@ -395,6 +395,10 @@ func (tp *TxProcessor) ProcessTxs(coordIdxs []common.Idx, l1usertxs, l1coordinat
 
 	// Process L2Txs
 	for i := 0; i < len(l2txs); i++ {
+		if l2txs[i].Type == common.TxTypeExit && l2txs[i].Amount.Cmp(big.NewInt(0)) == 0 {
```
Comment (jeffprestes) [inline txprocessor/txprocessor.go:398]: It seems to be OK.
However, is there a way to give any feedback to the user? Maybe block this in API level too?

**Review by jeffprestes — APPROVED** [review]

File: txprocessor/txprocessor.go (hunk 1 above)
Comment (Pantani) [inline txprocessor/txprocessor.go:398]: Is there any way to the amount be negative? Maybe we should filter this way:

```suggestion
		if l2txs[i].Type == common.TxTypeExit && l2txs[i].Amount.Cmp(big.NewInt(0)) <= 0 {
```

Comment (Pantani) [reply txprocessor/txprocessor.go:398]: Also, can we test this case?

**Comment by krlosMata** [comment]
Could you please double check on this scenario:
 - `txprocessor.go` is used in `sync` mode
 - assume that an alternative node process a L2 exit transaction with amount = 0
 -  Proper data-availability would be on the smart contract. In this case `From = X | To = 1(exit_idx) | Amount = 0  | Fee = X`
 - `hermez-node` should process this L2 exit transaction with amount = 0 to build the new state tree, since the sender account will increase its nonce by 1

**Comment by jeffprestes** [comment]
After @krlosMata comments, could you please , @arnaubennassar , test this scenario in sync mode?

**Review by jeffprestes — CHANGES_REQUESTED** [review]
Test sync mode with another node processing exit with amount 0 seems to be appropriated

File: txprocessor/txprocessor.go (hunk 1 above)
Comment (arnaubennassar) [reply txprocessor/txprocessor.go:398]: Marking as resolved as you already approved the PR that will do just do it

**Review by jeffprestes — COMMENTED** [review]
It seems to be OK. However, I strongly recommend to test this change in internaltestnet before to merge.

**Review by Pantani — APPROVED** [review]

**Comment by arnaubennassar** [comment]
> It seems to be OK. However, I strongly recommend to test this change in internaltestnet before to merge.

Tested successfully on internaltestnet

**Review by jeffprestes — APPROVED** [review]

**Pull request merged**